htmd/queues/localqueue.py

```diff
@@ -144,12 +183,37 @@ def inprogress(self):
 
         return output_run + output_queue
 
+    def notcompleted(self):
+        """Returns the sum of the number of job directories which do not have the sentinel file for completion.
+
+        Returns
+        -------
+        total : int
+            Total number of directories which have not completed
+        """
+        total = 0
+        if len(self._dirs) == 0:
+            raise RuntimeError('This method relies on running synchronously.')
+        for i in self._dirs:
+            if not os.path.exists(os.path.join(i, self._sentinel)):
+                total += 1
+        return total
+
     def stop(self):
         self._shutdown = True
 
-    @abc.abstractmethod
+    @abstractmethod
     def _getdevices(self):
-        pass
+        return list()
+
+    def _getmemory(self):
+        ureg = UnitRegistry()
+        total_memory = int(ureg.Quantity(psutil.virtual_memory().total, ureg.byte).to('MiB').magnitude)
+        nr_devices = len(self._getdevices())
+        if nr_devices != 0:
+            return int(total_memory/nr_devices)
```

Comment:
This is horribly nasty btw. It assumes the computer has 0 memory in use.

Reply:
virtual_memory().total is all total memory available, not only the available.

what do you propose?